src/controllers/twoFactorAuth.controller.ts

```diff
@@ -0,0 +1,62 @@
+import { Request, Response } from "express";
+import User from "../database/models/User.model";
+import * as twofactor from "node-2fa";
+import sendEmailToken from "../services/mails/sendEmailToken";
+
+/*
+  this middleware will enable two factor authentication on user account, create random code and send it to user through email and have to be verified to allow user access protected routes 
+*/
+export const create2FAToken = async (req: Request, res: Response) => {
+  try {
+    const newSecret = twofactor.generateSecret({
+      name: "Ecom-backend",
+      account: req.user.surname,
+    });
+
+    const user = await User.findByPk(req.user.id);
+
+    const code = twofactor.generateToken(newSecret.secret);
+
+    if (user && code && twofactor.verifyToken(newSecret.secret, code.token)) {
+      await user.update({
+        twoFAenabled: true,
+        twoFAverified: false,
+        secret: newSecret.secret,
+      });
+      res.send({ data: { ...newSecret, ...code }, status: 200 });
+      await sendEmailToken(user.email, code.token);
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      res.status(500).send({ error: error.message, status: 500 });
+    }
+  }
+};
```

Comment:
```suggestion
export const create2FAToken = async (req: Request, res: Response) => {
  const { id, surname } = req.user; // Try destructing your variables at the start of your function
  try {
    const { secrete } = twofactor.generateSecret({
      name: "Ecom-backend",
      account: surname,
    });

    const user = await User.findByPk(id);

    const { token }= twofactor.generateToken(secret);

    if (user)) {
      await user.update({
        twoFAenabled: true,
        twoFAverified: false,
        secret: secret,
      });
      await sendEmailToken(user.email, token);
      res.send({ data: { secret, token }, status: 200 });
    }
    else {
              throw new Error(`User not found.`)
    }
  } catch (error) {
    if (error instanceof Error) {
      res.status(500).send({ error: error.message, status: 500 });
    }
  }
};
```

Reply:
@musllim Try refactoring this function to the proposed one, looks more clean and is easily understood. 
implement the same logic for the function `verify2FAToken`, you have some `if` conditions that don't make sense at all.

Reply:
thanks